runtime/compiler/trj9/control/MethodToBeCompiled.cpp

```diff
@@ -50,7 +50,7 @@ TR_MethodToBeCompiled *TR_MethodToBeCompiled::allocate(J9JITConfig *jitConfig)
 
 void TR_MethodToBeCompiled::initialize(TR::IlGeneratorMethodDetails & details, void *oldStartPC, CompilationPriority p, TR_OptimizationPlan *optimizationPlan)
    {
-   _methodDetails = details;
+   _methodDetails = J9::IlGeneratorMethodDetails::create(_methodDetailsStorage, details);
```

Comment:
This should not referenced J9::IlGeneratorMethodDetails directly. Only TR::IlGeneratorMethodDetails should be referenced outside of its class hierarchy.

You'll need to expose a static create() call in the TR:: version that directs down to the J9 static implementation.

Reply:
I thought that for code within the `trj9` folder it was ok to refer use `J9::`, or is it because `IlGeneratorMethodDetails` is extensible that we have to use `TR::`. But then I see examples in CompilationThread.cpp of `J9::NewInstanceThunkDetails` instead of `TR::NewInstanceThunkDetails`.